### PR TITLE
[AUD-1008] EnablePushNotifications reminder drawer

### PR DESCRIPTION
**Pull request by sliptype**
### Description

Dependent on https://github.com/AudiusProject/audius-mobile-client/pull/134

This PR:
* Adds the EnablePushNotifications drawer
Simulator Screen Shot - iPhone 11 - 2021-10-28 at 13 52 30
* Adds a custom Button component
* Adds a custom Text component

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*
iOS and android devices, ios simulaor

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### CODE REVIEW

File: src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+import IconFollow from '../../assets/images/iconFollow.svg'
+import IconCoSign from '../../assets/images/iconCoSign.svg'
+
+import Button from '../../components/button'
+import Drawer from '../../components/drawer'
+import Text from '../../components/text'
+
+// Importing directly from audius-client temporarily until
+// settings page is migrated because we still need push notification logic to work
+// on settings page and it doesn't necessarily make sense in common
```
Comment (raymondjacobson) [inline src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx:19]: makes sense. lets capture this kind of dependency on linear or something

File: src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                style={styles.actionIcon}
+              />
+              <Text style={styles.actionText} weight='bold'>
+                {label}
+              </Text>
+            </View>
+          ))}
+        </View>
+        <Button
+          title='Enable Notifications'
```
Comment (raymondjacobson) [inline src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx:194]: should also prb be in messages

File: src/components/text/Text.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+
+import { Text as RNText } from 'react-native'
+import { useThemeColors } from '../../utils/theme'
+
+const fontByWeight = {
```
Comment (raymondjacobson) [inline src/components/text/Text.tsx:6]: nice. this is great

**Review by raymondjacobson — APPROVED** [review]
code lgtm! haven't seen the demo, but i can wait to test flight ;)

File: src/App.tsx [hunk 4]
```diff
@@ -71,6 +72,7 @@ const App = () => {
           */}
             {/* <BottomBar /> */}
             <MobileUploadDrawer />
```
[the remark this answers is not in the document]
Comment (sliptype) [reply src/App.tsx:74]: Agreed

File: src/assets/images/iconRemix copy.svg [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (sliptype) [reply src/assets/images/iconRemix copy.svg]: Oops, didn't notice I added a copy. Those icons already existed

File: src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+
+import { useDrawer } from '../../hooks/useDrawer'
+import { ThemeColors, useThemedStyles } from '../../hooks/useThemedStyles'
+import { useThemeColors } from '../../utils/theme'
+import LinearGradient from 'react-native-linear-gradient'
+import MaskedView from '@react-native-masked-view/masked-view'
+import { useDispatchWeb } from '../../hooks/useDispatchWeb'
+
+const messages = {
+  dontMiss: `Don't Miss a Beat!`,
```
[the remark this answers is not in the document]
Comment (sliptype) [reply src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx]: Done

File: src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx (hunk 1 above)
Comment (sliptype) [reply src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx:19]: Ok, noted in the settings page ticket

File: src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx (hunk 2 above)
Comment (sliptype) [reply src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx]: Done

**Review by raymondjacobson — APPROVED** [review]

**Pull request merged**